src/extension/wps/wps-core/src/main/java/org/geoserver/wps/validator/ObjectSizeEstimators.java

```diff
@@ -0,0 +1,51 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wps.validator;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.geoserver.platform.GeoServerExtensions;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+/**
+ * Central class providing static methods to estimate the size of an object. It will scan the
+ * application context looking for {@link ObjectSizeEstimator} objects
+ * 
+ * @author Andrea Aime - GeoSolutions
+ *
+ */
+public class ObjectSizeEstimators implements ApplicationContextAware {
```

Comment:
They are pluggable and declared in the spring context, so that people creating new processes with new/strange input objects can play in the size limiting game.